todo.tex

```diff
@@ -0,0 +1,24 @@
+\newpage
+\section{Proposed Improvements}\label{sec:proimp}
+
+Given the definitions in section \ref{sec:definitions} and the current status and problems described in appendix \ref{sec:statusAndProblems}, the following way forward is proposed.
+
+\begin{itemize}
+\item identify a software products that requires delivery soon.
+A possible candidate could be the \textit{Calibration} pipeline which is now  part of the Science Pipelines, to be used soon for AuxTel commissioning.
+The software product shall be defined using a \textit{metapackage} that includes only the Git packages relevant to it.
+Git packages that are shared with others software products, need to be included in a different metapackage,
+\item identify all the Git packages that are shared with other software products and included them in a \textit{library metapackage}.
+This library metapackage is a software product itself and needs to be managed and released separately.
+\item update the build system in order to be able to resolve the library from the binary repository (EUPS)
+\item update the tooling to deal with tagging (code-kit) and continuous integration (Jenkins scripts).
+\end{itemize}
+
+All the git packages included in a software product metapackage will be built and released all at the same time, as it is done now for the full Science Pipelines.
+
+A metapackage which depends on a library metapackage will resolve the library binaries from the EUPS binary repository.
+Dependencies between metapackages shall be handled differently, for example using a \textit{requirements.txt} file, than the usual EUPS requirements that are specified the the \textit{ups/gitpackagename.table}.
```

Comment:
I don't understand this. Does it imply that using EUPS to `setup Calibration` will not set up the “library metapackage” on which `Calibration` depends? If so, that's not acceptable.

Reply:
I am not expecting EUPS setup to be impacted. Here I am referring on the lsstsw rebuild, that is getting now all dependencies from GitHub. A metapackage shall be retrieved from eupspkgroot instead, if available.
This paragraph is redundant and better to remove it.